pennylane/circuit_graph.py

```diff
@@ -167,22 +167,29 @@ def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
 
         # TODO: State preparations demolish the incoming state entirely, and therefore should have no incoming edges.
 
-        self._graph = nx.DiGraph()  #: nx.DiGraph: DAG representation of the quantum circuit
+        self._graph = rx.PyDiGraph(
+            multigraph=False
+        )  #: rx.PyDiGraph: DAG representation of the quantum circuit
+
         # Iterate over each (populated) wire in the grid
         for wire in self._grid.values():
             # Add the first operator on the wire to the graph
             # This operator does not depend on any others
-            self._graph.add_node(wire[0])
+
+            if wire[0] not in self._graph.nodes():
+                self._graph.add_node(wire[0])
```

Comment:
How come you need to check the node doesn't exist in RX? Do you get an error otherwise?

Reply:
This is one of those few RX and NX differences that took me hours to find out about! If you don't check `wire[0]` is already added to the graph, RX will create a new node with the same value but a different index, while NX doesn't add any new node.

Reply:
nice catch in that case! Could you add a comment on this to the code? I imagine this might be something future developers wonder about

Reply:
Ooops! I thought that I already commented on this.